### DISCUSSION TITLE
Custom command examples

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2179] [Show and tell]
Just a  place to collect (more or less) useful bash commands that can be used in the "Custom Command" panel widget.

* Used partition space in /dev/sda1
`df -h|grep sda1|awk {'print $5'}`
On left click: "filelight" or similar tools

* Temperature
`acpi -t |awk {'print $4'}|sed 's/.0$//' `
or for PC:
` cat /sys/class/thermal/thermal_zone0/temp |cut -c1,2`
On left click: "Qps"

* Battery percentage:
`upower -i /org/freedesktop/UPower/devices/battery_BAT0|grep percent|awk {'print $2'}`
![schermata-01-13-11-11](https://user-images.githubusercontent.com/10681413/149312160-f3b25901-ed42-4d79-8836-a021e6e48f78.png)

* Battery remaining time:
`upower -i /org/freedesktop/UPower/devices/battery_BAT0|grep time|awk {'print $4,$5'}`
`

**Comment by stefonarch** [comment]
* Free memory percentage:
`free | grep Mem | awk '{print $7/$2 * 100.0}'|cut -d '.' -f 1`

**Reply by tsujan** [reply]
We may need a Wiki page for lxqt-panel. These interesting commands may not be easy to find here.

**Comment by luis-pereira** [comment]
> * Free partition space in /dev/sda1
>   `df -h|grep sda1|awk {'print $5'}`

This is **used** partition space, not free space.

**Reply by stefonarch** [reply]
Corrected, thanks!

**Comment by stefonarch** [comment]
As the custom command plugin supports now images here are some examples:
https://github.com/niknah/panel-customcommand-scripts

**Reply by Gnomasz** [reply]
Those do not work for me and I have no idea if I've set up something wrong or if there's a bug? The xml tags aren't parsed, but shown as-is:
![obraz](https://github.com/lxqt/lxqt/assets/58573071/ba5d516f-7639-403e-b6d0-d80aab7878c1)
This is the setup:
![obraz](https://github.com/lxqt/lxqt/assets/58573071/c8b25907-f0c0-4aad-aba1-0979b393ad8b)
The script is executable and trusted.
Am I missing something?

**Reply by tsujan** [reply]
> Those do not work for me

Because you don't have https://github.com/lxqt/lxqt-panel/commit/13e827320d9e0ab2b43cb14080bc4695fda832c9. LXQt Panel 1.4.0 will have it and will be released tomorrow.

**Comment by stefonarch** [comment]
Network speed:

![immagine](https://github.com/user-attachments/assets/0c97c1ca-734f-464e-ad37-dd78625dade3)


`ifstat | grep enp2s0f3u2u4 | awk '{print "⬇ "$2" KB/s ⬆ "$3" KB/s"}'`

(see https://github.com/lxqt/lxqt/discussions/2600)

Bandwidth/month:

![immagine](https://github.com/user-attachments/assets/6f2302da-6a36-4258-85fa-8fede61096b0)

`vnstat -m|grep estim|awk '{print $8 $9}'

**Comment by elviosak** [comment]
Analog clock (structured output)

https://github.com/user-attachments/assets/03081625-fadb-437b-90ab-746e93087e90

``` bash
time=$(date +"%H:%M:%S")
IFS=":"
read hour minute second <<< "$time"
hour=$(( 10#$hour ))
minute=$(( 10#$minute ))
second=$(( 10#$second ))
icon=$(echo -n "<svg viewBox='0 0 100 100'>
  <circle cx='50' cy='50' r='50' fill='#ddeeff' />
  <path
  stroke='#999999'
  stroke-width='5'
  d='M0,50 h20 M50,0 v20 M100,50 h-20 M50,100 v-20'/>
  <path
    transform='rotate($(( hour * 30 + minute / 2 )) 50 50)'
    fill='none'
    stroke='#111166'
    stroke-width='16'
    stroke-linecap='round'
    d='M50,56 v-40' />
  <path
    transform='rotate($(( minute * 6 + second / 10 )) 50 50)'
    fill='none'
    stroke='#226622'
    stroke-width='12'
    stroke-linecap='round'
    d='M50,56 v-49' />
  <path
    transform='rotate($(( second * 6 )) 50 50)'
    fill='none'
    stroke='#ff3333'
    stroke-width='8'
    stroke-linecap='round'
    d='M50,56 v-51' />
</svg>" | base64 -w0)
IFS=" "
time=$(echo -n "$time" | base64 -w 0)
echo "icon:$icon text:$time"
```

**Reply by tsujan** [reply]
Very nice!

**Comment by linuxissuper** [comment]
GPU usage   
<img width="175" height="33" alt="image" src="https://github.com/user-attachments/assets/49f54fb4-c628-401b-9e58-21aabc37e023" />
  
`printf "%02d%s%s%s%02d%s" $(cat /sys/class/drm/card1/device/gpu_busy_percent) " % | 🌡 "  $(sensors | grep amdgpu-pci-0300 --after-context=4 | grep edge: | awk '{print $2}') "  | 🌬 " $(sensors | grep amdgpu-pci-0300 --after-context=4 | grep fan1: | awk '{print $2}') ""`   
  
needs adjustments:
`card1` is maybe `card0`  
replace `amdgpu-pci-0300` with your gpu name (see `sensors` output)


<img width="556" height="583" alt="image" src="https://github.com/user-attachments/assets/589f95c7-1894-465c-8455-0259f44238c1" />
